src/catalog/src/durable/upgrade/persist/v47_to_v48.rs

```diff
@@ -0,0 +1,790 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use mz_stash::upgrade::WireCompatible;
+use mz_stash::wire_compatible;
+use std::collections::BTreeMap;
+
+use crate::durable::upgrade::persist::MigrationAction;
+use crate::durable::upgrade::{objects_v47 as v47, objects_v48 as v48};
+
+wire_compatible!(v47::DatabaseKey with v48::DatabaseKey);
+wire_compatible!(v47::SchemaKey with v48::SchemaKey);
+wire_compatible!(v47::RoleKey with v48::RoleKey);
+wire_compatible!(v47::ItemKey with v48::ItemKey);
+wire_compatible!(v47::RoleId with v48::RoleId);
+wire_compatible!(v47::MzAclItem with v48::MzAclItem);
+wire_compatible!(v47::DatabaseId with v48::DatabaseId);
+wire_compatible!(v47::RoleAttributes with v48::RoleAttributes);
+wire_compatible!(v47::RoleMembership with v48::RoleMembership);
+wire_compatible!(v47::RoleVars with v48::RoleVars);
+wire_compatible!(v47::SchemaId with v48::SchemaId);
+wire_compatible!(v47::ClusterIntrospectionSourceIndexKey with v48::ClusterIntrospectionSourceIndexKey);
+wire_compatible!(v47::CatalogItem with v48::CatalogItem);
+
+/// Persist OIDs in the catalog.
+pub fn upgrade(
+    snapshot: Vec<v47::StateUpdateKind>,
+) -> Vec<MigrationAction<v47::StateUpdateKind, v48::StateUpdateKind>> {
+    let mut cur_user_oid: u32 = 20_000;
```

Comment:
Slight preference to wrap this in a newtype which automatically increments the value when you get an ID, but up to you

Reply:
Another good idea, updated.